packages/editor-ui/src/Interface.ts

```diff
@@ -1159,7 +1162,7 @@ export interface UIState {
 	sidebarMenuCollapsed: boolean;
 	modalStack: string[];
 	modals: {
-		[key: string]: IModalState;
+		[key: string]: IModalState | NewCredentialsModal;
```

Comment:
types could get more specific
```suggestion
    newModal: NewCredentialsModal;
		[key: string]: IModalState;
```

Reply:
Agree, added new types and typeguards for this.